ipaserver/install/ipa_custodia_regen.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (C) 2017  IPA Project Contributors, see COPYING for license
+"""Regenerate Custodia server.keys and config
+
+KEEP THIS SCRIPT COMPATIBLE WITH FREEIPA 4.5
+
+Christian Heimes <cheimes@redhat.com>
+"""
+from __future__ import print_function
+
+import logging
+import os
+
+from ipalib import api
+from ipaplatform.paths import paths
+from ipapython import admintool
+from ipapython.ipautil import backup_file
+from ipapython.ipa_log_manager import standard_logging_setup
+from ipaserver.install import installutils
+from ipaserver.install import sysupgrade
+from ipaserver.install.custodiainstance import CustodiaInstance
+
+
+logger = logging.getLogger(__name__)
+
+
+class IPACustodiaRegen(admintool.AdminTool):
+    command_name = 'ipa-custodia-regen'
+
+    usage = "%prog [options]\n"
+
+    @classmethod
+    def add_options(cls, parser):
+        super(IPACustodiaRegen, cls).add_options(parser)
+        parser.add_option(
+            "-r", '--regenerate', action="store_true",
```

Comment:
What is being regenreated with this option ?
The tool is called ipa-custodia-*regen* already, what should the user expect by running it with or without this option, why is it necessary ?

Reply:
The script does nothing without an option. It's a poor man's version of **read the documentation first**.